addons/portal/controllers/mail.py

```diff
@@ -98,8 +98,8 @@ def portal_chatter_post(self, res_model, res_id, message, redirect=None, attachm
 
         res_id = int(res_id)
 
-        attachment_ids = [int(res_id) for res_id in attachment_ids.split(',')]
-        attachment_tokens = attachment_tokens.split(',')
+        attachment_ids = [int(res_id) for res_id in attachment_ids.split(',') if res_id]
```

Comment:
doesn't it shadow the variable `res_id` defined above? (may depend on python version)

Reply:
No idea, it seemed to work on 3.5 and 3.6... But to be safe I renamed the two variables.